### ISSUE TITLE
Update process is confusing

### ISSUE THREAD

**Issue by eltang** [UltimateHackingKeyboard/agent#597]
It was not clear to me that pressing the Update button starts the download of the new AppImage. While I did notice the new file and try to execute it, it was incomplete due to me having closed Agent prematurely.

**Comment by mondalaci** [comment]
Admittedly, these are very early days regarding the update process. What has exactly happened on your side?

I clicked on the "Check for udpate" link and a spinner appeared. Then I saw the following in the terminal:

```
[23:30:19.891] [debug] [ '[AppUpdateEffect] call checkForUpdate' ]
[23:30:19.893] [debug] [AppUpdateService] checkForUpdate request from renderer process
[23:30:19.894] [info] firstRunVersion: 1.1.2
[23:30:19.894] [info] package.version: 1.1.2
[23:30:19.905] [info] Checking for update
[23:30:19.906] [info] sendIpcToWindow: checking-for-update undefined
[23:30:21.566] [info] Found version 1.1.3 (url: https://github.com/UltimateHackingKeyboard/agent/releases/download/v1.1.3/UHK.Agent-1.1.3-linux-x86_64.AppImage)
[23:30:21.568] [info] Downloading update from https://github.com/UltimateHackingKeyboard/agent/releases/download/v1.1.3/UHK.Agent-1.1.3-linux-x86_64.AppImage
[23:30:21.574] [info] Downloading update from https://github.com/UltimateHackingKeyboard/agent/releases/download/v1.1.3/UHK.Agent-1.1.3-linux-x86_64.AppImage
[23:30:21.578] [debug] [AppUpdateService] checkForUpdate success
[23:30:29.156] [error] Cannot download differentially, fallback to full download: Error: unexpected end of file
    at Zlib._handle.onerror (zlib.js:355:17)
From previous event:
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/differentialPackage.js:22:1
    at Generator.next (<anonymous>)
From previous event:
    at readBlockMap (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/differentialPackage.js:26:1)
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/differentialPackage.js:105:1
From previous event:
    at DifferentialDownloader.downloadAppImage (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/differentialPackage.js:107:10)
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppImageUpdater.js:120:26
    at Generator.next (<anonymous>)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
From previous event:
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppImageUpdater.js:132:1
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/BaseUpdater.js:85:1
    at Generator.next (<anonymous>)
    at FSReqWrap.oncomplete (fs.js:114:15)
From previous event:
    at AppImageUpdater.executeDownload (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/BaseUpdater.js:95:10)
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppImageUpdater.js:105:1
    at Generator.next (<anonymous>)
From previous event:
    at AppImageUpdater.doDownloadUpdate (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppImageUpdater.js:139:10)
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppUpdater.js:347:1
    at Generator.next (<anonymous>)
From previous event:
    at AppImageUpdater.downloadUpdate (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppUpdater.js:352:10)
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppUpdater.js:322:1
From previous event:
    at AppImageUpdater.doCheckForUpdates (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppUpdater.js:324:10)
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppUpdater.js:272:1
    at Generator.next (<anonymous>)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
From previous event:
    at AppImageUpdater._checkForUpdates (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppUpdater.js:277:10)
    at AppImageUpdater.checkForUpdates (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppUpdater.js:223:1)
    at AppUpdateService.checkForUpdate (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/src/services/app-update.service.ts:90:21)
    at EventEmitter.electron_1.ipcMain.on (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/src/services/app-update.service.ts:70:18)
    at emitTwo (events.js:106:13)
    at EventEmitter.emit (events.js:194:7)
    at WebContents.<anonymous> (/tmp/.mount_UHK.Agi6R0mj/app/resources/electron.asar/browser/api/web-contents.js:247:13)
    at emitTwo (events.js:106:13)
    at WebContents.emit (events.js:194:7)
[23:30:30.485] [info] sendIpcToWindow: auto-update-download-progress { total: 53605146,
  delta: 6785823,
  transferred: 6785823,
  percent: 12.658902188233942,
  bytesPerSecond: 6607423 }
[23:30:31.512] [info] sendIpcToWindow: auto-update-download-progress { total: 53605146,
  delta: 11321272,
  transferred: 18107095,
  percent: 33.778650654174136,
  bytesPerSecond: 8815528 }
[23:30:31.636] [error] Cannot download differentially, fallback to full download: Error: unexpected end of file
    at Zlib._handle.onerror (zlib.js:355:17)
From previous event:
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/differentialPackage.js:22:1
    at Generator.next (<anonymous>)
From previous event:
    at readBlockMap (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/differentialPackage.js:26:1)
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/differentialPackage.js:105:1
From previous event:
    at DifferentialDownloader.downloadAppImage (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/differentialPackage.js:107:10)
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppImageUpdater.js:120:26
    at Generator.next (<anonymous>)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
From previous event:
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppImageUpdater.js:132:1
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/BaseUpdater.js:85:1
    at Generator.next (<anonymous>)
    at FSReqWrap.oncomplete (fs.js:114:15)
From previous event:
    at AppImageUpdater.executeDownload (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/BaseUpdater.js:95:10)
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppImageUpdater.js:105:1
    at Generator.next (<anonymous>)
From previous event:
    at AppImageUpdater.doDownloadUpdate (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppImageUpdater.js:139:10)
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppUpdater.js:347:1
    at Generator.next (<anonymous>)
From previous event:
    at AppImageUpdater.downloadUpdate (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppUpdater.js:352:10)
    at AppUpdateService.<anonymous> (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/src/services/app-update.service.ts:31:31)
    at Generator.next (<anonymous>)
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/electron-main.js:23637:71
    at Promise (<anonymous>)
    at __awaiter (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/electron-main.js:23633:12)
    at AppImageUpdater.electron_updater_1.autoUpdater.on (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/src/services/app-update.service.ts:30:76)
    at emitOne (events.js:96:13)
    at AppImageUpdater.emit (events.js:191:7)
    at AppImageUpdater.onUpdateAvailable (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppUpdater.js:328:1)
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppUpdater.js:315:1
From previous event:
    at AppImageUpdater.doCheckForUpdates (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppUpdater.js:324:10)
    at /tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppUpdater.js:272:1
    at Generator.next (<anonymous>)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
From previous event:
    at AppImageUpdater._checkForUpdates (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppUpdater.js:277:10)
    at AppImageUpdater.checkForUpdates (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/out/AppUpdater.js:223:1)
    at AppUpdateService.checkForUpdate (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/src/services/app-update.service.ts:90:21)
    at EventEmitter.electron_1.ipcMain.on (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/src/services/app-update.service.ts:70:18)
    at emitTwo (events.js:106:13)
    at EventEmitter.emit (events.js:194:7)
    at WebContents.<anonymous> (/tmp/.mount_UHK.Agi6R0mj/app/resources/electron.asar/browser/api/web-contents.js:247:13)
    at emitTwo (events.js:106:13)
    at WebContents.emit (events.js:194:7)
[23:30:32.542] [info] sendIpcToWindow: auto-update-download-progress { total: 53605146,
  delta: 12353488,
  transferred: 30460583,
  percent: 56.82399036838739,
  bytesPerSecond: 9876972 }
[23:30:32.992] [info] sendIpcToWindow: auto-update-download-progress { total: 53605146,
  delta: 5426479,
  transferred: 5426479,
  percent: 10.123056096144202,
  bytesPerSecond: 5232863 }
[23:30:33.565] [info] sendIpcToWindow: auto-update-download-progress { total: 53605146,
  delta: 12894520,
  transferred: 43355103,
  percent: 80.87862124281874,
  bytesPerSecond: 10556392 }
[23:30:34.029] [info] sendIpcToWindow: auto-update-download-progress { total: 53605146,
  delta: 5603568,
  transferred: 11030047,
  percent: 20.576470400808162,
  bytesPerSecond: 5318248 }
[23:30:34.286] [info] sendIpcToWindow: auto-update-download-progress { total: 53605146,
  delta: 10250043,
  transferred: 53605146,
  percent: 100,
  bytesPerSecond: 11102971 }
[23:30:34.288] [info] New version 1.1.3 has been downloaded to /tmp/up-mqfGjj/UHK.Agent-1.1.3-linux-x86_64.AppImage
[23:30:34.289] [info] sendIpcToWindow: update-downloaded undefined
[23:30:35.068] [info] sendIpcToWindow: auto-update-download-progress { total: 53605146,
  delta: 5635904,
  transferred: 16665951,
  percent: 31.090207272264497,
  bytesPerSecond: 5353662 }
[23:30:36.108] [info] sendIpcToWindow: auto-update-download-progress { total: 53605146,
  delta: 5636288,
  transferred: 22302239,
  percent: 41.60466049285642,
  bytesPerSecond: 5370151 }
[23:30:37.146] [info] sendIpcToWindow: auto-update-download-progress { total: 53605146,
  delta: 5635904,
  transferred: 27938143,
  percent: 52.11839736431275,
  bytesPerSecond: 5382035 }
[23:30:38.186] [info] sendIpcToWindow: auto-update-download-progress { total: 53605146,
  delta: 5685368,
  transferred: 33623511,
  percent: 62.72440895879661,
  bytesPerSecond: 5396166 }
[23:30:39.225] [info] sendIpcToWindow: auto-update-download-progress { total: 53605146,
  delta: 5816248,
  transferred: 39439759,
  percent: 73.5745762169923,
  bytesPerSecond: 5425001 }
[23:30:40.264] [info] sendIpcToWindow: auto-update-download-progress { total: 53605146,
  delta: 6029312,
  transferred: 45469071,
  percent: 84.82221277785533,
  bytesPerSecond: 5472268 }
[23:30:41.303] [info] sendIpcToWindow: auto-update-download-progress { total: 53605146,
  delta: 6390000,
  transferred: 51859071,
  percent: 96.74271011219706,
  bytesPerSecond: 5547611 }
[23:30:41.547] [info] sendIpcToWindow: auto-update-download-progress { total: 53605146,
  delta: 1746075,
  transferred: 53605146,
  percent: 100,
  bytesPerSecond: 5588526 }
[23:30:41.549] [info] New version 1.1.3 has been downloaded to /tmp/up-bGbCa8/UHK.Agent-1.1.3-linux-x86_64.AppImage
[23:30:41.549] [info] sendIpcToWindow: update-downloaded undefined
[23:30:41.550] [info] sendIpcToWindow: update-available undefined
```

At this point, "New version available. [Update button] [Close button]" (CSS should be improved) appeared on the top. I clicked on the Update button and got the following in the terminal:

```
[23:34:47.748] [debug] [AppUpdateService] update and restart from renderer process
[23:34:47.833] [info] Auto install update on quit
[23:34:47.833] [info] [Electron Main] win closed
[23:34:47.834] [info] [DeviceService] Device connection checker stopped.
[23:34:47.834] [debug] [UhkHidDevice] Device communication closing.
module.js:472
    throw err;
    ^

Error: Cannot find module '../dialog'
    at Module._resolveFilename (module.js:470:15)
    at Function.Module._resolveFilename (/tmp/.mount_UHK.Agi6R0mj/app/resources/electron.asar/common/reset-search-paths.js:35:12)
    at Function.Module._load (module.js:418:25)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.get [as dialog] (/tmp/.mount_UHK.Agi6R0mj/app/resources/electron.asar/browser/api/exports/electron.js:11:16)
    at process.<anonymous> (/tmp/.mount_UHK.Agi6R0mj/app/resources/electron.asar/browser/init.js:54:31)
    at emitOne (events.js:96:13)
    at process.emit (events.js:191:7)
    at process.emit (/tmp/.mount_UHK.Agi6R0mj/app/resources/app.asar/webpack:/~/electron-updater/~/source-map-support/source-map-support.js:439:1)
    at process._fatalException (bootstrap_node.js:308:26)
```

At this point Agent terminated.

**Comment by ert78gb** [comment]
@mondalaci do you try to upgrade from 1.1.2 to 1.1.3?

**Comment by ert78gb** [comment]
@mondalaci could you check the os processes for any Agent or electron processes?

**Comment by mondalaci** [comment]
Yes, I tried to update from 1.1.2 to 1.1.3.

There were no other Agent/electron processes running while running Agent.

I'm able to run multiple Agent processes at the same time. This is not a requirement. As a matter of fact, it'd be better if the user could only run one Agent process at the same time.

**Comment by ert78gb** [comment]
Maybe the electron-userland/electron-builder#2745 fix solved this issue. The fix was released on 26th March with electron builder 20.8.1 release and it has upgraded in agent in 31th March that is mean it only available from Agent 1.1.3.

@mondalaci Could you create an new alpha release and run the update from 1.1.3 to 1.1.4?

**Comment by mondalaci** [comment]
@ert78gb Sure thing. I'll create a normal release on Monday. It'll make sense anyways because you've made improvements since the last release.

**Comment by ert78gb** [comment]
I plan to finish #467 today

**Comment by mondalaci** [comment]
@ert78gb Just released 1.1.4. I used 1.1.3 and tried to update to 1.1.4. The following happened:

```
[15:07:08.205] [debug] [ '[AppUpdateEffect] call checkForUpdate' ]
[15:07:08.206] [debug] [AppUpdateService] checkForUpdate request from renderer process
[15:07:08.207] [info] firstRunVersion: 1.1.3
[15:07:08.207] [info] package.version: 1.1.3
[15:07:08.208] [info] Checking for update
[15:07:08.208] [info] sendIpcToWindow: checking-for-update undefined
[15:07:10.170] [info] Found version 1.1.4 (url: UHK.Agent-1.1.4-linux-x86_64.AppImage)
[15:07:10.172] [info] Downloading update from UHK.Agent-1.1.4-linux-x86_64.AppImage
[15:07:10.174] [info] Downloading update from UHK.Agent-1.1.4-linux-x86_64.AppImage
[15:07:10.176] [debug] [AppUpdateService] checkForUpdate success
[15:07:10.180] [info] No cached update available
[15:07:10.181] [info] No cached update available
[15:07:11.084] [info] File has 231 changed blocks
[15:07:11.084] [debug] [
  {
    "kind": 0,
    "start": 0,
    "end": 82292
  },
  {
    "kind": 1,
    "start": 82292,
    "end": 115060
  },
  {
    "kind": 0,
    "start": 115060,
    "end": 16549572
  },
  {
    "kind": 1,
    "start": 16549572,
    "end": 19116636
  },
  {
    "kind": 0,
    "start": 19120797,
    "end": 19131322
  },
  {
    "kind": 1,
    "start": 19127161,
    "end": 21161865
  },
  {
    "kind": 0,
    "start": 21141823,
    "end": 53331162
  },
  {
    "kind": 1,
    "start": 53351204,
    "end": 53402281
  },
  {
    "kind": 0,
    "start": 53382240,
    "end": 53490747
  },
  {
    "kind": 1,
    "start": 53510788,
    "end": 53667512
  }
]
[15:07:11.095] [info] Full: 52,464.78 KB, To download: 4,728.84 KB (9%)
[15:07:11.098] [info] Differential download: https://github.com/UltimateHackingKeyboard/agent/releases/download/v1.1.4/UHK.Agent-1.1.4-linux-x86_64.AppImage
[15:07:11.102] [debug] effective url: , range: bytes=82292-115059
[15:07:11.291] [info] Redirect to https://github-production-release-asset-2e65be.s3.amazonaws.com/49464167/721ff50a-3c02-11e8-8ac5-84849afc8ad3
[15:07:11.433] [info] File has 231 changed blocks
[15:07:11.433] [debug] [
  {
    "kind": 0,
    "start": 0,
    "end": 82292
  },
  {
    "kind": 1,
    "start": 82292,
    "end": 115060
  },
  {
    "kind": 0,
    "start": 115060,
    "end": 16549572
  },
  {
    "kind": 1,
    "start": 16549572,
    "end": 19116636
  },
  {
    "kind": 0,
    "start": 19120797,
    "end": 19131322
  },
  {
    "kind": 1,
    "start": 19127161,
    "end": 21161865
  },
  {
    "kind": 0,
    "start": 21141823,
    "end": 53331162
  },
  {
    "kind": 1,
    "start": 53351204,
    "end": 53402281
  },
  {
    "kind": 0,
    "start": 53382240,
    "end": 53490747
  },
  {
    "kind": 1,
    "start": 53510788,
    "end": 53667512
  }
]
[15:07:11.433] [info] Full: 52,464.78 KB, To download: 4,728.84 KB (9%)
[15:07:11.435] [info] Differential download: https://github.com/UltimateHackingKeyboard/agent/releases/download/v1.1.4/UHK.Agent-1.1.4-linux-x86_64.AppImage
[15:07:11.437] [debug] effective url: , range: bytes=82292-115059
[15:07:11.635] [info] Redirect to https://github-production-release-asset-2e65be.s3.amazonaws.com/49464167/721ff50a-3c02-11e8-8ac5-84849afc8ad3
[15:07:11.784] [debug] effective url: https://github-production-release-asset-2e65be.s3.amazonaws.com/49464167/721ff50a-3c02-11e8-8ac5-84849afc8ad3, range: bytes=16549572-19116635
[15:07:12.013] [debug] effective url: https://github-production-release-asset-2e65be.s3.amazonaws.com/49464167/721ff50a-3c02-11e8-8ac5-84849afc8ad3, range: bytes=16549572-19116635
[15:07:13.988] [debug] effective url: https://github-production-release-asset-2e65be.s3.amazonaws.com/49464167/721ff50a-3c02-11e8-8ac5-84849afc8ad3, range: bytes=19127161-21161864
[15:07:13.991] [debug] effective url: https://github-production-release-asset-2e65be.s3.amazonaws.com/49464167/721ff50a-3c02-11e8-8ac5-84849afc8ad3, range: bytes=19127161-21161864
[15:07:14.756] [debug] effective url: https://github-production-release-asset-2e65be.s3.amazonaws.com/49464167/721ff50a-3c02-11e8-8ac5-84849afc8ad3, range: bytes=53351204-53402280
[15:07:14.922] [debug] effective url: https://github-production-release-asset-2e65be.s3.amazonaws.com/49464167/721ff50a-3c02-11e8-8ac5-84849afc8ad3, range: bytes=53510788-53667511
[15:07:15.071] [info] New version 1.1.4 has been downloaded to /home/laci/.config/uhk-agent/UHK.Agent-1.1.4-linux-x86_64.AppImage
[15:07:15.071] [info] sendIpcToWindow: update-downloaded undefined
[15:07:15.071] [info] sendIpcToWindow: update-available undefined
[15:07:15.165] [debug] effective url: https://github-production-release-asset-2e65be.s3.amazonaws.com/49464167/721ff50a-3c02-11e8-8ac5-84849afc8ad3, range: bytes=53351204-53402280
[15:07:15.438] [debug] effective url: https://github-production-release-asset-2e65be.s3.amazonaws.com/49464167/721ff50a-3c02-11e8-8ac5-84849afc8ad3, range: bytes=53510788-53667511
[15:07:15.711] [error] Error: Error: ENOENT: no such file or directory, chmod '/home/laci/.config/uhk-agent/temp-UHK.Agent-1.1.4-linux-x86_64.AppImage'
    at task (/tmp/.mount_UHK.Ag4OXzfp/app/resources/app.asar/webpack:/home/travis/build/UltimateHackingKeyboard/agent/~/electron-updater/out/AppImageUpdater.js:155:1)
    at <anonymous>
[15:07:15.711] [error] [AppUpdateService] error Error: ENOENT: no such file or directory, chmod '/home/laci/.config/uhk-agent/temp-UHK.Agent-1.1.4-linux-x86_64.AppImage'
    at task (/tmp/.mount_UHK.Ag4OXzfp/app/resources/app.asar/webpack:/home/travis/build/UltimateHackingKeyboard/agent/~/electron-updater/out/AppImageUpdater.js:155:1)
    at <anonymous>
[15:07:15.711] [info] sendIpcToWindow: auto-update-error Error: ENOENT: no such file or directory, chmod '/home/laci/.config/uhk-agent/temp-UHK.Agent-1.1.4-l
(node:1641) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: ENOENT: no such file or directory, chmod '/home/laci/.config/uhk-agent/temp-UHK.Agent-1.1.4-linux-x86_64.AppImage'
(node:1641) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

At this point, I clicked on the Update button and got this:

```
[15:07:30.361] [debug] [AppUpdateService] update and restart from renderer process
[15:07:30.362] [info] Install on explicit quitAndInstall
[15:07:30.365] [error] Error: Error: No valid update available, can't quit and install
    at AppImageUpdater.install (/tmp/.mount_UHK.Ag4OXzfp/app/resources/app.asar/webpack:/home/travis/build/UltimateHackingKeyboard/agent/~/electron-updater/out/BaseUpdater.js:155:1)
    at AppImageUpdater.quitAndInstall (/tmp/.mount_UHK.Ag4OXzfp/app/resources/app.asar/webpack:/home/travis/build/UltimateHackingKeyboard/agent/~/electron-updater/out/BaseUpdater.js:62:1)
    at EventEmitter.electron_1.ipcMain.on (/tmp/.mount_UHK.Ag4OXzfp/app/resources/app.asar/webpack:/src/services/app-update.service.ts:59:32)
    at emitTwo (events.js:125:13)
    at EventEmitter.emit (events.js:213:7)
    at WebContents.<anonymous> (/tmp/.mount_UHK.Ag4OXzfp/app/resources/electron.asar/browser/api/web-contents.js:266:13)
    at emitTwo (events.js:125:13)
    at WebContents.emit (events.js:213:7)
[15:07:30.366] [error] [AppUpdateService] error Error: No valid update available, can't quit and install
    at AppImageUpdater.install (/tmp/.mount_UHK.Ag4OXzfp/app/resources/app.asar/webpack:/home/travis/build/UltimateHackingKeyboard/agent/~/electron-updater/out/BaseUpdater.js:155:1)
    at AppImageUpdater.quitAndInstall (/tmp/.mount_UHK.Ag4OXzfp/app/resources/app.asar/webpack:/home/travis/build/UltimateHackingKeyboard/agent/~/electron-updater/out/BaseUpdater.js:62:1)
    at EventEmitter.electron_1.ipcMain.on (/tmp/.mount_UHK.Ag4OXzfp/app/resources/app.asar/webpack:/src/services/app-update.service.ts:59:32)
    at emitTwo (events.js:125:13)
    at EventEmitter.emit (events.js:213:7)
    at WebContents.<anonymous> (/tmp/.mount_UHK.Ag4OXzfp/app/resources/electron.asar/browser/api/web-contents.js:266:13)
    at emitTwo (events.js:125:13)
    at WebContents.emit (events.js:213:7)
[15:07:30.366] [info] sendIpcToWindow: auto-update-error Error: No valid update available, can't quit and install
    at AppImageUpdater.install (/tmp/.mount
```

**Comment by mondalaci** [comment]
This is an old issue, and much of its content is not relevant anymore. Nevertheless, the auto update feature is buggy, so I'll open another issue and closing this one.
